daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -1104,7 +1106,9 @@ func (devices *DeviceSet) initDevmapper(doInit bool) error {
 
 	// https://github.com/docker/docker/issues/4036
 	if supported := devicemapper.UdevSetSyncSupport(true); !supported {
-		logrus.Warn("Udev sync is not supported. This will lead to unexpected behavior, data loss and errors. For more information, see https://docs.docker.com/reference/commandline/cli/#daemon-storage-driver-option")
+		if !devices.overrideUdevSyncCheck {
+			logrus.Warn("Udev sync is not supported. This will lead to unexpected behavior, data loss and errors. For more information, see https://docs.docker.com/reference/commandline/cli/#daemon-storage-driver-option")
```

Comment:
I think what @vbatts means is that the `if !devices.overrideUdevSyncCheck {` check here is not necessary. I don't think it makes a big difference and I believe we can keep it, but maybe I'm missing something.